infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/OperationalPointDao.kt

```diff
@@ -147,6 +154,45 @@ class OperationalPointDao(
 
     @Transactional fun save(item: OperationalPoint): LayoutRowVersion<OperationalPoint> = save(item, NoParams.instance)
 
+    @Transactional
+    fun insertRatkoPoint(id: IntId<OperationalPoint>, ratkoPointVersion: Int): LayoutRowVersion<OperationalPoint> {
+        val sql =
+            """
+                insert into
+                  layout.operational_point(
+                    id,
+                    draft,
+                    design_id,
+                    layout_context_id,
+                    state,
+                    origin,
+                    ratko_operational_point_version
+                )
+                values (
+                  :id,
+                  true,
+                  null,
+                  'main_draft',
```

Comment:
Niin layout_context_id taidettiin meillä insertoida erikseen? Ei sinällään vain tämän pullarin asiaa mutta eikös tuon kannattaisi olla generoitu column, koska ei se kuitenkaan sisällä tietoa joka ei jo muissa columneissa olisi.

Reply:
Muistelisin että siinä oli jotain suht monimutkaista postgres-hähmää. Suoraan tältä istumalta en osaa sanoa muuta kuin että se ei ainakaan sopinut ihan heti yhteen versiotaulutriggereiden kanssa:

> Generated columns are, conceptually, updated after BEFORE triggers have run. Therefore, changes made to base columns in a BEFORE trigger will be reflected in generated columns. But conversely, it is not allowed to access generated columns in BEFORE triggers.

... tosin en muistaakseni myöskään katsonut sitten pidemmälle tätä että saisiko sen toimimaan siirtämällä triggerit myöhemmäksi. Tällähän ratkaisulla kun ei tarvitse miettiä lainkaan, missä järjestyksessä asiat tapahtuvat postgressin sisälmyksissä.

Reply:
Ok, no tuo selittää. Versioinnissahan oli tosiaan kaksi triggeriä: toinen päivittää pääriville tiedot ja toinen kopioi rivin versiotauluun. Niiden järjestys oli aika herkkä jotta nuo tapahtuu oikein päin + deletoinnit kirjautuu oikein. En muista olisiko ne siirrettävissä mutta ainakin potentiaalisen hazardi.

.. tietenkin nuohan voisi myös täydentää triggerillä generoidun columnin sijaan, mutta sille pitäisi melkein sitten olla oma triggerinsä jne joten ruma varmaan sekin. Noh, pitääpä miettiä. Tuntuisi vähän asialta joka saisi tulla automaagisesti mutta ei tuo kovin ison vaivan arvoinen asia ole.